### PR TITLE
date: add module

### PR DESCRIPTION
**Pull request by michamos**
These utils are used to handle correctly partial dates, lacking day or
month parts.

Signed-off-by: Micha Moskovic <michamos@gmail.com>

### CODE REVIEW

File: inspire_utils/date.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        return 'PartialDate(year={self.year}, month={self.month}, day={self.day})'.format(self=self)
+
+    def __eq__(self, other):
+        return self.year == other.year and self.month == other.month and self.day == other.day
+
+    def __lt__(self, other):
+        other_month = other.month or 99
+        other_day = other.day or 99
+
+        return self.dumps() < self._dumps(other.year, other_month, other_day)
```
Comment (kaplun) [inline inspire_utils/date.py]: Actually you can use tuples here and inline `_dumps` in `dumps`

**Comment by michamos** [comment]
@jacquerie ready for review (don't have rights to ask for it in the Github way).

File: inspire_utils/date.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        """Pretty print the date.
+
+        Examples:
+            >>> PartialDate(1686, 6, 30).pprint()
+            u'Jun 30, 1686'
+        """
+        if not self.month:
+            return dates.format_date(datetime.date(self.year, 1, 1), 'YYYY', locale='en')
+        if not self.day:
+            return dates.format_date(datetime.date(self.year, self.month, 1), 'MMM, YYYY', locale='en')
```
Comment (michamos) [inline inspire_utils/date.py]: the locale is probably not needed in Inspire, but it was needed on my laptop that is set-up in Italian.

**Comment by kaplun** [comment]
@michamos Just fixed the global rights. Now every member of `inspirehep` can push/pull and tweak tickets. I am sure everybody is responsible enough not to do any mess.

**Comment by jacquerie** [comment]
> I am sure everybody is responsible enough not to do any mess.

...he said, after setting the wrong permissions : )

**Comment by kaplun** [comment]
ha

**Comment by kaplun** [comment]
I haven't set the permission of this package. I haven't even created it. I have now set global permission of inspire organization so that by default new repo will authorize everybody to propose tickets and push and pulls. I don't see what I've done wrong.

**Comment by jacquerie** [comment]
> I don't see what I've done wrong.

I see that you added rights to the group `content-tool-admins`, which contains you and @michamos...

**Comment by kaplun** [comment]
Yes this was the initial way to fix the rights for Micha, but then I also amended generally as described above, which indeed makes the `content-tool-admins` redundant. (or almost since it also grant admin rights on this repo to @michamos, which I believe is an OK thing, since this is an auxiliary repo).

**Pull request merged**